### PR TITLE
Dash/datagrid edit modified

### PR DESCRIPTION
**Pull request by pawellysy**
Implemented disabling of editing of the columns, which are created by Math Modifier.

### CODE REVIEW

**Review by sebastianbochan — APPROVED** [review]

File: ts/Dashboards/Plugins/DataGridComponent.ts [hunk 1]
```diff
@@ -24,12 +24,16 @@ import type DataGrid from '../../DataGrid/DataGrid';
 import type DataTable from '../../Data/DataTable';
 import type DataGridOptions from '../../DataGrid/DataGridOptions';
 import type BaseDataGridOptions from '../../DataGrid/DataGridOptions';
```
Comment (karolkolodziej) [inline ts/Dashboards/Plugins/DataGridComponent.ts:26]: Do we need these two imports? Looks like we import the same thing

File: ts/Dashboards/Plugins/DataGridComponent.ts [hunk 2]
```diff
@@ -227,6 +236,36 @@ class DataGridComponent extends Component {
...
+     * @param connector
+     * Attached connector
+     */
+    private disableEditingModifiedColumns(connector: DataConnectorType) {
+        const dataModifier = connector.options.dataModifier;
+
+        if (!dataModifier || dataModifier.type !== 'Math') {
+            return;
+        }
+        const modifierColumns = (dataModifier as MathModifierOptions).columnFormulas;
```
Comment (karolkolodziej) [inline ts/Dashboards/Plugins/DataGridComponent.ts:252]: Most likely we don't need this casting.

File: ts/Dashboards/Plugins/HighchartsComponent.ts [hunk 3]
```diff
@@ -602,7 +604,8 @@ class HighchartsComponent extends Component {
                     .modifyTable(this.presentationTable).modified;
             }
 
-            const table = this.presentationTable;
+            const table = this.presentationTable,
+                modifierOoptions = table.getModifier()?.options;
```
Comment (karolkolodziej) [inline ts/Dashboards/Plugins/HighchartsComponent.ts:608]: typo

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/Dashboards/Plugins/DataGridComponent.ts [hunk 4]
```diff
@@ -22,7 +22,6 @@
 import type Cell from '../Layout/Cell';
 import type DataGrid from '../../DataGrid/DataGrid';
 import type DataTable from '../../Data/DataTable';
-import type DataGridOptions from '../../DataGrid/DataGridOptions';
 import type BaseDataGridOptions from '../../DataGrid/DataGridOptions';
```
Comment (karolkolodziej) [inline ts/Dashboards/Plugins/DataGridComponent.ts:25]: I'm voting in favour of `DataGridOptions` instead of `BaseDataGridOptions`

Comment (pawellysy) [reply ts/Dashboards/Plugins/DataGridComponent.ts:25]: There is a name conflict, so maybe let's leave the `BaseDataGridOptions` :)

Comment (karolkolodziej) [reply ts/Dashboards/Plugins/DataGridComponent.ts:25]: ah, i didn't see that ;)

**Review by karolkolodziej — APPROVED** [review]

File: samples/dashboards/demo/datagrid-mathmodifier/demo.html [hunk 5]
```diff
@@ -7,12 +7,12 @@
...
-16, 23, 1.0881
-17, 15, 1.0829
-18, 27, 1.0813
-19, 13, 1.0808</pre>
+<pre id="csv">Day,EUR,Rate
+15,11,1.0876
+16,23,1.0881
+17,15,1.0829
+18,27,1.0813
+19,13,1.0808</pre>
```
Comment (pawelfus) [inline samples/dashboards/demo/datagrid-mathmodifier/demo.html:15]: Why changing this? 🤔

File: ts/Dashboards/Plugins/HighchartsComponent.ts [hunk 6]
```diff
@@ -656,9 +659,22 @@ class HighchartsComponent extends Component {
...
+                        (formula) => formula.column == seriesName
+                    )
+                );
+
                 return chart.addSeries({
                     name: seriesName,
-                    id: `${storeTableID}-series-${index}`
+                    id: `${storeTableID}-series-${index}`,
+                    dragDrop: {
+                        draggableY: shouldBeDraggable
```
Comment (pawelfus) [inline ts/Dashboards/Plugins/HighchartsComponent.ts:676]: We need this info in API - any thoughts on where we should place it? Not in Highcharts API obviously.

Perhaps in the general description of the Chart component? We could list default options for the series (and the chart) there.

**Review by pawelfus — CHANGES_REQUESTED** [review]

File: samples/dashboards/demo/datagrid-mathmodifier/demo.html (hunk 5 above)
Comment (pawellysy) [reply samples/dashboards/demo/datagrid-mathmodifier/demo.html:15]: with spaces, the name of the series will be ` EUR` with the extra space in the beginning, which may cause some bugs, when we will try to test something in the future. I thought that a good idea would be to fix it here by the way.

**Review by pawelfus — APPROVED** [review]
LGTM

**Pull request merged**